.circleci/config.yml

```diff
@@ -0,0 +1,168 @@
+---
+version: 2.1
+
+parameters:
+  # This is the branch that this pipeline will build. Set this to your
+  # "master" branch for your repository.
+  build-branch:
+    type: string
+    default: "v2.8.0-minim"
+
+workflows:
+  build:
+    # Uncomment the filters to only run builds for this on the main branch
+    jobs:
+      # Set the version and make the build agent
+      - prepare_cimg:
+          context: deployable-service
+          filters:
+            branches:
+              only:
+                - << pipeline.parameters.build-branch >>
+      # Build openWRT configurations
+      - build:
+          requires:
+            - prepare_cimg
+          # Adjust the matrix parameters to add/remove devices
+          matrix:
+            parameters:
+              device: ["motorola_q14", "motorola_mh7020", "motorola_r14"]
+          context: deployable-service
+          # Add more branches to these filter blocks if you want to build more branches
+          # They include the "master branch" from the parameter above by default
+          filters:
+            branches:
+              only:
+                - << pipeline.parameters.build-branch >>
+      # Push builds to the releases feed
+      - promote:
+          name: promote-<< matrix.device >>
+          requires:
+            - build-<< matrix.device >>
+          matrix:
+            parameters:
+              device: ["motorola_q14", "motorola_mh7020", "motorola_r14"]
+          context: minimos
+          filters:
+            branches:
+              only:
+                - << pipeline.parameters.build-branch >>
+      # Tag the wlan-ap and unum-sdk repos with the version
+      - tag:
+          name: tag
+          requires: 
+            - promote
+          filters:
+            branches:
+              only:
+                - << pipeline.parameters.build-branch >>
+
+# Every job is a new filesystem
+# Every "run" step is a new shell
+jobs:
+  prepare_cimg:
+    docker:
+      - image: cimg/base:2023.02
+    working_directory: ~/project
+    steps:
+      - checkout
+      # Build a docker image with all the openwrt prereqs for the next steps
+      - run: sudo apt-get update && sudo apt-get install awscli -y && sudo apt-get clean
+      # Make sure to use a remote docker version in-support with CircleCI
+      # https://circleci.com/docs/building-docker-images/#docker-version
+      - setup_remote_docker:
+          version: 20.10.18
+          docker_layer_caching: true
+      # This script uses AWS CLI v1 (installed above). 
+      # We could use cimg/aws runner (AWS CLI v2) if we improved the ECR login in this script.
+      - run: ./bin/circleci_buildciimg.sh
+      # Create the version here and make it available to remaining jobs via persist_to_workspace
+      - run: echo $(cat base_version).$CIRCLE_BUILD_NUM > ./version
```

Comment:
This step could check if a tag exists on the current commit, and if it does, use that tag as the build number instead of constructing one.